automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -365,19 +369,23 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
 
   def withNewErroredCluster[T](googleProject: GoogleProject)(testCode: Cluster => T)(implicit token: AuthToken): T = {
     val name = ClusterName(s"automation-test-a${makeRandomId()}z")
-    val request = ClusterRequest(Map("foo" -> makeRandomId()), Some(incorrectJupyterExtensionUri))
-    val testResult: Try[T] = Try {
-      val cluster = createAndMonitor(googleProject, name, request)
-      cluster.status shouldBe ClusterStatus.Error
-      cluster.errors should have size 1
-      cluster.errors.head.errorMessage should include ("gs://")
-      cluster.errors.head.errorCode should be (3)
-      testCode(cluster)
-    }
+    // Fail a cluster by providing a user script which returns exit status 1
+    val hailUploadFile = ResourceFile("bucket-tests/invalid_user_script.sh")
+    withResourceFileInBucket(googleProject, hailUploadFile, "text/plain") { bucketPath =>
+      val request = ClusterRequest(jupyterUserScriptUri = Some(bucketPath.toUri))
+      val testResult: Try[T] = Try {
+        val cluster = createAndMonitor(googleProject, name, request)
+        cluster.status shouldBe ClusterStatus.Error
```

Comment:
probably is outside of scope of this PR, assertions are normally done in test. it feels weird to have assertion in a `util` class.

Reply:
I think it's helpful sometimes to have assertions in non-test code, too if they 

1. make an implicit assumption explicit for self-documenting
2. print a message that makes troubleshooting easier in case of failure
2. make the test/app fail sooner where it wouldn't make sense for the execution to continue